sdk/nodejs/runtime/settings.ts

```diff
@@ -379,7 +381,7 @@ export function _setStack(val: string | undefined) {
  * Returns true if we are currently connected to a resource monitoring service.
  */
 export function hasMonitor(): boolean {
-    return !!monitor && !!options().monitorAddr;
+    return (!!monitor && !!options().monitorAddr) || options().monitorAddr === "mock";
```

Comment:
Should this look at `getStore` rather than `options()` to see if `monitor` is set? Would match the check for the global `monitor`